sstables/storage.cc

```diff
@@ -461,21 +461,20 @@ class s3_storage : public sstables::storage {
 };
 
 sstring s3_storage::make_s3_object_name(const sstable& sst, component_type type) {
-    return format("/{}/{}/{}", _bucket, *_remote_prefix, sstable_version_constants::get_component_map(sst.get_version()).at(type));
+    return format("/{}/{}/{}", _bucket, sst.generation(), sstable_version_constants::get_component_map(sst.get_version()).at(type));
 }
 
 future<> s3_storage::ensure_remote_prefix(const sstable& sst) {
     if (!_remote_prefix) {
-        auto uuid = co_await sst.manager().system_keyspace().sstables_registry_lookup_entry(_location, sst.generation());
-        _remote_prefix = uuid.to_sstring();
+        co_await sst.manager().system_keyspace().sstables_registry_lookup_entry(_location, sst.generation());
+        _remote_prefix = fmt::to_string(sst.generation());
     }
 }
 
 void s3_storage::open(sstable& sst) {
-    auto uuid = utils::UUID_gen::get_time_UUID();
     entry_descriptor desc("", "", "", sst._generation, sst._version, sst._format, component_type::TOC);
-    sst.manager().system_keyspace().sstables_registry_create_entry(_location, uuid, status_creating, std::move(desc)).get();
-    _remote_prefix = uuid.to_sstring();
+    sst.manager().system_keyspace().sstables_registry_create_entry(_location, status_creating, std::move(desc)).get();
+    _remote_prefix = fmt::to_string(sst._generation);
```

Comment:
It should somehow be checked or enforced that the generation is uuid-one, not the legacy integer. Otherwise we may end up putting an `s3://bucket/number/basename` object on S3 potentially corrupting someone else's same-name one without knowing it early enough

Reply:
thank you for pointing out this. it makes a lot sense! i am still thinking about where shall we do the validation. i think, there are some combinations of configurations are invalid or does not make sense at all. and the sooner we identify them and error out the better.

Reply:
i just check it in `s3_storage::make_s3_object_name()`.